src/main/java/com/aws/greengrass/util/S3SdkClientFactory.java

```diff
@@ -89,7 +89,7 @@ public S3Client getS3Client() throws DeviceConfigurationException {
      * @param r region
      * @return s3client
      */
-    public S3Client getClientForRegion(Region r) {
+    public synchronized S3Client getClientForRegion(Region r) {
```

Comment:
this does not need to be synchronized. We are correctly using a concurrent hashmap